File: storage/storage-mysql/src/main/java/io/say/better/storage/mysql/domain/entity/Record.java
package io.say.better.storage.mysql.domain.entity;

import io.say.better.storage.mysql.domain.constant.ReactionType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Getter
@Entity(name = "RECORD")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Record {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "record_id")
	private Long recordId;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "review_id", nullable = false)
	private Review review;

	@Column(name = "order_num", nullable = false, columnDefinition = "int default 0")
	private Integer orderNum;

	@Enumerated(EnumType.STRING)
	@Column(name = "reaction_type", nullable = false, length = 20)
	private ReactionType reactionType;

}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domain/entity/Record.kt
package io.say.better.storage.mysql.domain.entity

import io.say.better.storage.mysql.domain.constant.ReactionType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity(name = "RECORD")
class Record(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "record_id")
    private var recordId: Long = 0,
    @JoinColumn(name = "review_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    private val review: Review,
    @Column(name = "order_num", nullable = false, columnDefinition = "int default 0")
    private var orderNum: Int,
    @Enumerated(EnumType.STRING)
    @Column(name = "reaction_type", length = 20)
    private var reactionType: ReactionType? = null,
)
